test/common/json/test_data/generate_test_data.py

```diff
@@ -0,0 +1,339 @@
+        data=blob,
+        throws=True,
+    )
+
+    blob = get_blob(HTTP_CONN_NETWORK_FILTER_BLOB)
+    blob['access_log'].append({"path": "/dev/null", "filter": {"type": "logical_or", "filters": []}})
+    write_test_file(
+        'LessThanTwoFiltersInListNoneLogicalOrThrows',
+        schema='HTTP_CONN_NETWORK_FILTER_SCHEMA',
+        data=blob,
+        throws=True,
+    )
+
+    blob = get_blob(HTTP_CONN_NETWORK_FILTER_BLOB)
+    blob['access_log'].append({"path": "/dev/null", "filter": {"type": "logical_and", "filters": []}})
+    write_test_file(
+        'LessThanTwoFiltersInListNoneLogicalAndThrows',
+        schema='HTTP_CONN_NETWORK_FILTER_SCHEMA',
+        data=blob,
+        throws=True,
+    )
+
+    blob = get_blob(HTTP_CONN_NETWORK_FILTER_BLOB)
+    blob['access_log'].append({"path": "/dev/null", "filter": {"type": "logical_or", "filters": [{"type": "not_healthcheck"}]}})
+    write_test_file(
+        'LessThanTwoFiltersInListOneLogicalOrThrows',
+        schema='HTTP_CONN_NETWORK_FILTER_SCHEMA',
+        data=blob,
+        throws=True,
+    )
+
+    blob = get_blob(HTTP_CONN_NETWORK_FILTER_BLOB)
+    blob['access_log'].append({"path": "/dev/null", "filter": {"type": "logical_and", "filters": [{"type": "not_healthcheck"}]}})
+    write_test_file(
+        'LessThanTwoFiltersInListOneLogicalAndThrows',
+        schema='HTTP_CONN_NETWORK_FILTER_SCHEMA',
+        data=blob,
+        throws=True,
+    )
+
+    write_test_file(
+        'Valid',
+        schema='ROUTE_CONFIGURATION_SCHEMA',
+        data=get_blob(ROUTE_CONFIGURATION_BLOB),
+        throws=False,
+    )
+
+    write_test_file(
+        'Valid',
+        schema='ROUTE_ENTRY_CONFIGURATION_SCHEMA',
+        data=get_blob(ROUTE_ENTRY_CONFIGURATION_BLOB),
+        throws=False,
+    )
+
+    blob = {"prefix": "/foo", "cluster": "local_service_grpc", "priority": "foo"}
+    write_test_file(
+        'InvalidPriority',
+        schema='ROUTE_ENTRY_CONFIGURATION_SCHEMA',
+        data=blob,
+        throws=True,
+    )
+
+    write_test_file(
+        'Valid',
+        schema='CLUSTER_SCHEMA',
+        data=get_blob(CLUSTER_BLOB),
+        throws=False,
+    )
+
+    blob = get_blob(CLUSTER_BLOB)
+    blob['features'] = "nonexistentfeature"
+    write_test_file(
+        'UnsupportedFeature',
+        schema='CLUSTER_SCHEMA',
+        data=blob,
+        throws=True,
+    )
+
+    write_test_file(
+        'Valid',
+        schema='TOP_LEVEL_CONFIG_SCHEMA',
+        data=get_blob(TOP_LEVEL_CONFIG_BLOB),
+        throws=False,
+    )
+
+    blob = get_blob(TOP_LEVEL_CONFIG_BLOB)
+    blob['tracing']['http']['driver']['type'] = 'unknown'
+    write_test_file(
+        'UnsupportedTracingDriver',
+        schema='TOP_LEVEL_CONFIG_SCHEMA',
+        data=blob,
+        throws=True,
+    )
+
+    
+def get_blob(blob):
+    return copy.deepcopy(blob)
+
+def write_test_file(name, schema, data, throws):
+    test_filename = os.path.join(
+        os.environ['TEST_TMPDIR'],
```

Comment:
To avoid clashing in the coverage test, can we use a subdir of TEST_TMPDIR. E.g., ${TEST_TMPDIR}/json_schema_test.